src/tar.ts

```diff
@@ -53,36 +67,40 @@ async function getTarVersion(): Promise<TarVersion> {
 
 export interface ZstdAvailability {
   available: boolean;
+  foundZstdBinary: boolean;
   version?: TarVersion;
 }
 
 export async function isZstdAvailable(
   logger: Logger,
 ): Promise<ZstdAvailability> {
+  const foundZstdBinary = await isBinaryAccessible("zstd", logger);
   try {
     const tarVersion = await getTarVersion();
     const { type, version } = tarVersion;
     logger.info(`Found ${type} tar version ${version}.`);
     switch (type) {
       case "gnu":
         return {
-          available: version >= MIN_REQUIRED_GNU_TAR_VERSION,
+          available: foundZstdBinary && version >= MIN_REQUIRED_GNU_TAR_VERSION,
+          foundZstdBinary,
           version: tarVersion,
         };
       case "bsd":
         return {
-          available: version >= MIN_REQUIRED_BSD_TAR_VERSION,
+          available: foundZstdBinary && version >= MIN_REQUIRED_BSD_TAR_VERSION,
+          foundZstdBinary,
           version: tarVersion,
         };
       default:
         assertNever(type);
     }
   } catch (e) {
-    logger.error(
+    logger.warning(
       "Failed to determine tar version, therefore will assume zstd may not be available. " +
```

Comment:
Minor: I think this is clearer.

```suggestion
      "Failed to determine tar version, therefore will assume zstd is not available. " +
```